air/src/air/mod.rs

```diff
@@ -579,25 +582,59 @@ pub trait Air: Send + Sync {
         R: RandomCoin<BaseField = Self::BaseField>,
     {
         let mut t_coefficients = Vec::new();
-        for _ in 0..self.trace_info().width() {
-            t_coefficients.push(public_coin.draw()?);
+        match self.context().options.batching_type_deep() {
+            BatchingType::Linear => {
+                for _ in 0..self.trace_info().width() {
+                    t_coefficients.push(public_coin.draw()?);
+                }
+
+                let mut c_coefficients = Vec::new();
+                for _ in 0..self.context().num_constraint_composition_columns() {
+                    c_coefficients.push(public_coin.draw()?);
+                }
+
+                let lagrange_cc = if self.context().has_lagrange_kernel_aux_column() {
+                    Some(public_coin.draw()?)
+                } else {
+                    None
+                };
```

Comment:
nit: I would maybe move this logic into methods on `DeepCompositionCoefficients`. For example, something like:

```rust
impl<E: FieldElement> DeepCompositionCoefficients<E> {
    pub draw_linear(
        public_coin: &mut impl RandomCoin<BaseField = E::BaseField>,
        num_values: usize,
    ) -> Self {
        ...
    }

    pub draw_algebraic(
        public_coin: &mut impl RandomCoin<BaseField = E::BaseField>,
        num_values: usize,
    ) -> Self {
        ...
    }
}
```

And then the code here becomes:
```rust
match self.context().options.deep_batching() {
    BatchingType::Linear => DeepCompositionCoefficients::draw_linear(),
    BatchingType::Algebraic => DeepCompositionCoefficients::draw_algebraic(),
}
```

Reply:
The issue is that `num_values` is not enough, we need the dimensions of the committed traces

Reply:
Ah - I see! I guess we could pass in the individual dimensions - e.g., like:
```rust
impl<E: FieldElement> DeepCompositionCoefficients<E> {
    pub draw_linear(
        public_coin: &mut impl RandomCoin<BaseField = E::BaseField>,
        trace_width: usize,
        num_constraint_composition_columns: usize,
    ) -> Self {
        ...
    }

    pub draw_algebraic(
        public_coin: &mut impl RandomCoin<BaseField = E::BaseField>,
        trace_width: usize,
        num_constraint_composition_columns: usize,
    ) -> Self {
        ...
    }
}
```
I'm skipping the Lagrange column info as we'll be removing it.

Reply:
Yeah, that would work. I can change to this